### ISSUE TITLE
Bug: Pasting a table that includes a 'caption' tag inside crashes the editor

### ISSUE THREAD

**Issue by ivailop7** [facebook/lexical#6910]
Lexical 0.21

Copy the example table that has a `<caption>` tag inside from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption and paste in Playground. Editor crashes.

**Comment by etrepum** [comment]
This is because the importDOM doesn't throw away the caption node. I think something like this at the end of `$convertTableElement` should work around this issue

```ts
  return {
    // TODO - we may want to support <caption> as the first node, eventually
    after: (childLexicalNodes) => childLexicalNodes.filter($isTableRowNode),
    node: tableNode,
  };
```

**Comment by etrepum** [comment]
TableNode is still a bit of a mess with unnecessary and probably incorrect indexing (considering rowSpan/colSpan) that would need fixing in a lot of places especially if the caption node is to be supported, but this should prevent the crashing on import. An additional measure to make things a bit more robust is to add a TableNode transform to remove all children that are not TableRowNode (and probably do the same for TableRowNode to remove all children that are not TableCellNode).

**Comment by accessabilly** [comment]
It seems like `<thead>`, `<tbody>` and `<tfoot>` crash tables, too.

**Comment by ivailop7** [comment]
> TableNode is still a bit of a mess with unnecessary and probably incorrect indexing (considering rowSpan/colSpan) that would need fixing in a lot of places especially if the caption node is to be supported, but this should prevent the crashing on import. An additional measure to make things a bit more robust is to add a TableNode transform to remove all children that are not TableRowNode (and probably do the same for TableRowNode to remove all children that are not TableCellNode).

Yeah, I'm with you on this. I don't think we need to support `caption`. My report was more on preventing the crash cause. If people want to override the TableNode and implement their own support for 'caption', they can do so.

**Comment by accessabilly** [comment]
As Lexical [prominently claims](https://lexical.dev/) to be accessible, I would deny that. The `<caption>` element is used to describe complex data tables and it works like a directly associated heading for it. If a table has no caption it can fail a success criterion "[1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG22/#info-and-relationships)" of WCAG 2.1 and create inaccessible content for that reason.

Read more here: https://www.w3.org/WAI/WCAG22/Techniques/html/H39

**Comment by etrepum** [comment]
The accessibility claim is for the core package and refers to it being able to leverage most of the accessibility features built-in to the browser via contentEditable, unlike other rich text editors such as its predecessor Draft.js. Accessibility in the DOM input/output is really up to the user's implementation, lexical is a framework, not a drop-in full featured editor. Tables are not part of the core and are still very much in development, they barely worked at all until recently. You're welcome to contribute PRs to add caption support if your application needs them sooner than later.
